### DISCUSSION TITLE
Will the Granite model(s) ever support function calling?

### DISCUSSION THREAD

**Discussion by jamesfalkner** [ibm-granite/discussions#11] [General]
Just curious.. it seems like a very powerful feature to extend the model. ala https://platform.openai.com/docs/guides/function-calling

**Comment by mayank31398** [comment]
Hey, yeah very soon
we are working actively on function calling support.
This will take time though.

**Comment by mayank31398** [comment]
Hi @jamesfalkner 
this is available now: https://huggingface.co/ibm-granite/granite-20b-functioncalling
paper: https://arxiv.org/abs/2407.00121v1
